src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,129 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
-    @Override
-    public void add(T value) {
+    private static final int DEFAULT_SIZE = 10;
+    private Object[] values;
+    private int size;
+    private int arraySize = DEFAULT_SIZE;
+    private T removedObject;
```

Comment:
we don't need these fields

Reply:
We need removedObject field because we need to return removed(index) object in remove method